integration-test/src/test/java/org/sagebionetworks/ITAccessTokenTest.java

```diff
@@ -0,0 +1,147 @@
+		adminSynapse.clearAllLocks();
+		synapseOne = new SynapseClientImpl();
+		SynapseClientHelper.setEndpoints(synapseOne);
+		user1ToDelete = SynapseClientHelper.createUser(adminSynapse, synapseOne);
+
+		synapseAnonymous = new SynapseClientImpl();
+		SynapseClientHelper.setEndpoints(synapseAnonymous);
+	}
+
+	@AfterAll
+	public static void afterClass() throws Exception {
+		try {
+			if (user1ToDelete!=null) adminSynapse.deleteUser(user1ToDelete);
+		} catch (SynapseException e) { }
+		try {
+			if (user2ToDelete!=null) adminSynapse.deleteUser(user2ToDelete);
+		} catch (SynapseException e) { }
+	}
+
+	@BeforeEach
+	public void before() throws Exception {
+		// create the OAuth client
+		client = new OAuthClient();
+		client.setClient_name(UUID.randomUUID().toString());
+		client.setRedirect_uris(Collections.singletonList("https://foo.bar.com"));
+		client = synapseOne.createOAuthClient(client);
+		// Sets the verified status of the client (only admins and ACT can do this)
+		client = adminSynapse.updateOAuthClientVerifiedStatus(client.getClient_id(), client.getEtag(), true);
+		secret = synapseOne.createOAuthClientSecret(client.getClient_id()).getClient_secret();
+	}
+
+	@AfterEach
+	public void after() throws Exception {
+		try {
+			if (project!=null) adminSynapse.deleteEntity(project);
+		} catch (SynapseException e) { }
+		try {
+			if (client!=null) {
+				synapseOne.deleteOAuthClient(client.getClient_id());
+			}
+		} catch (SynapseException e) {
+			e.printStackTrace();
+		}
+	}
+
+	private String getAccessToken(String scopes) throws Exception {
+		OIDCAuthorizationRequest authorizationRequest = new OIDCAuthorizationRequest();
+		authorizationRequest.setClientId(client.getClient_id());
+		authorizationRequest.setRedirectUri(client.getRedirect_uris().get(0));
+		authorizationRequest.setResponseType(OAuthResponseType.code);
+		authorizationRequest.setScope(scopes);
+		authorizationRequest.setClaims("{\"id_token\":{},\"userinfo\":{}}");
+		String nonce = UUID.randomUUID().toString();
+		authorizationRequest.setNonce(nonce);		
+		OAuthAuthorizationResponse oauthAuthorizationResponse = synapseOne.authorizeClient(authorizationRequest);
+
+		// Note, we use Basic auth to authorize the client when asking for an access token
+		OIDCTokenResponse tokenResponse = null;
+		try {
+			synapseAnonymous.setBasicAuthorizationCredentials(client.getClient_id(), secret);
+			tokenResponse = synapseAnonymous.getTokenResponse(OAuthGrantType.authorization_code, 
+					oauthAuthorizationResponse.getAccess_code(), client.getRedirect_uris().get(0), null, null, null);
+			return tokenResponse.getAccess_token();
+		} finally {
+			synapseAnonymous.removeAuthorizationHeader();
+		}
+
+	}
+
+
+	@Test
+	public void testAccessToken() throws Exception {
+		String accessToken = getAccessToken("openid modify view download");
+
+		try {
+			// We use the bearer token to authorize the client 
+			synapseAnonymous.setBearerAuthorizationToken(accessToken);
```

Comment:
It's not really anonymous user as the the accessToken created above we use use a specific user

Reply:
Done.  I renamed multiple variables to clarify how we are using access tokens with the Synapse Java client.